src/utilities/new_versions.jl

```diff
@@ -99,3 +99,213 @@ function skip_equality_specifiers(
            !contains(compat_entry_verbatim, '>') &&
            !contains(compat_entry_verbatim, '<')
 end
+
+function create_new_pull_request(
+    api::GitLab.GitLabAPI,
+    repo::Gitlab.Project,
+    new_branch_name::AbstractString,
+    master_branch_name::AbstractString,
+    title::AbstractString,
+    body::AbstractString,
+)
+    return create_pull_request(
+        api,
+        repo.owner.name,
+        repo.name;
+        id=repo.id,
+        source_branch=new_branch_name,
+        target_branch=master_branch_name,
+        title=title,
+        body=body,
+    )
+end
+
+function create_new_pull_request(
+    api::GitHub.GutHubAPI,
+    repo::GitHub.Repo,
+    new_branch_name::AbstractString,
+    master_branch_name::AbstractString,
+    title::AbstractString,
+    body::AbstractString,
+)
+    return create_pull_request(
+        api,
+        repo.owner.login,
+        repo.name;
+        id=repo.id,
+        source_branch=new_branch_name,
+        target_branch=master_branch_name,
+        title=title,
+        body=body,
+    )
+end
+
+function get_url_with_auth(
+    api::GitHub.GitHubAPI, hostname::AbstractString, repo::GitHub.Repo
+)
+    return "https://x-access-token:$(api.token.token)@$hostname/$(repo.full_name).git"
+end
+
+function get_url_for_ssh(::GitHub.GitHubAPI, hostname::AbstractString, repo::GitHub.Repo)
+    return "git@$hostname:$(repo.full_name).git"
+end
+
+function get_url_with_auth(
+    api::GitLab.GitLabAPI, hostname::AbstractString, repo::GitLab.Project
+)
+    return "https://oauth2:$(api.token.token)@$hostname/$(repo.path_with_namespace).git"
+end
+
+function get_url_for_ssh(::GitLab.GitLabAPI, hostname::AbstractString, repo::GitLab.Project)
+    return "git@$(hostname):$(repo.path_with_namespace).git"
+end
+
+function make_pr_for_new_version(
```

Comment:
This function should really be piece-mealed apart it's way too long.

Reply:
yeah fair, I wanted to initially just merge all 3 into 1, and from here I can split it up a bit more into more logical chunks rather than the same function signature 3 times. This also changes it to that you need to cycle though the deps before calling "make_pr_for_new_version" instead of having one of them just cycle through the deps